### ISSUE TITLE
failed update: failed to find latest fs-repo-migrations: net/http: TLS handshake timeout 

### ISSUE THREAD

**Issue by yagamiraku** [ipfs/ipfs-desktop#1773]
Please describe what you were doing when this error happened.

**Specifications**

- **OS**: win32
- **IPFS Desktop Version**: 0.14.0
- **Electron Version**: 11.2.3
- **Chrome Version**: 87.0.4280.141

**Error**

```
TypeError: ctx.manualCheckForUpdates is not a function
    at click (E:\Program Files\IPFS Desktop\resources\app.asar\src\tray.js:216:30)
    at MenuItem.click (electron/js2c/browser_init.js:73:1683)
    at Object.a._executeCommand (electron/js2c/browser_init.js:81:2343)
```

**Comment by lidel** [comment]
@yagamiraku are you able to attach logs?

1. Find `IPFS Desktop/*.log` files (usually `C:/Windows/Users/$user/AppData/Roaming/IPFS Desktop/`)
2. Attach `error.log` and `combined.log` to this issue.

**Comment by yagamiraku** [comment]
[error.log](https://github.com/ipfs-shipyard/ipfs-desktop/files/6053729/error.log)

**Comment by lidel** [comment]
Thank you! I see it fails because it is unable to fetch `fs-repo-migrations`:

```
2021-02-27T04:26:15.323Z error: [ipfsd] start daemon Error: Initializing daemon...
go-ipfs version: 0.8.0
Repo version: 11
System version: amd64/windows
Golang version: go1.15.8
Found outdated fs-repo, migrations need to be run.
  => Looking for suitable fs-repo-migrations binary.
  => None found, downloading.
  => Failed to download fs-repo-migrations.
The migrations of fs-repo failed:
  failed to find latest fs-repo-migrations: http.DefaultClient.Do error: Get "https://ipfs.io/ipfs/QmYRLRDKobvg1AXTGeK5Xk6ntWTsjGiHbyNKhWfz7koGpa/fs-repo-migrations/versions": net/http: TLS handshake timeout
If you think this is a bug, please file an issue and include this whole log output.
  https://github.com/ipfs/fs-repo-migrations
 

 Error: failed to find latest fs-repo-migrations: http.DefaultClient.Do error: Get "https://ipfs.io/ipfs/QmYRLRDKobvg1AXTGeK5Xk6ntWTsjGiHbyNKhWfz7koGpa/fs-repo-migrations/versions": net/http: TLS handshake timeout 
```

I believe this is a new variant of https://github.com/ipfs-shipyard/ipfs-desktop/issues/1576 but ot fails at TLS handshake level.

@yagamiraku 
Are you running any firewall/antivirus software than could be responsible for intercepting web traffic?
Or behind something like The Great Firewall of China?

**Comment by yagamiraku** [comment]
Great Firewall, Yes.
I can't open that version URL with out proxy,
Have any way to use socks5 proxy?

Thank you for your patience

**Comment by lidel** [comment]
@yagamiraku sadly no built-in socks5 support in ipfs-desktop nor go-ipfs at this time. We are determined to improve the upgrade mechanism, so it does not break for users behind the Great Firewall, but for now you need to fix it manually.

I think the way to fix things for you is to:
1. Download  [fs-repo-migrations_v1.7.1_windows-amd64.zip](https://ipfs.io/ipfs/QmYRLRDKobvg1AXTGeK5Xk6ntWTsjGiHbyNKhWfz7koGpa/fs-repo-migrations/v1.7.1/fs-repo-migrations_v1.7.1_windows-amd64.zip)  using some proxy
2.  Add unpacked  `fs-repo-migrations.exe`   to your `PATH` – on Windows it means copying `.exe`  to `C:\Windows\system32\` or [manually adding some other directory via this interface](https://web.archive.org/web/20201223191048/https://www.architectryan.com/2018/03/17/add-to-the-path-on-windows-10/)
3. Quit IPFS Desktop and start it again

Let us know if this worked.

**Comment by yagamiraku** [comment]
It worked, thank you.

**Comment by lidel** [comment]
FYSA  go-ipfs 0.9.0 will ship soon  and will include a fix  (https://github.com/ipfs/go-ipfs/pull/8064)  that leverages IPFS itself for  fetching all the necessary `fs-repo-migrations`. Let's keep this issue open until ipfs-desktop ships with it.

**Comment by lidel** [comment]
Migration issue should be solved since [v0.16.0](https://github.com/ipfs/ipfs-desktop/releases/tag/v0.16.0) which includes go-ipfs 0.9.0 that will attempt to fetch migration data over IPFS.

**Comment by timecool-cpu** [comment]
> @yagamiraku sadly no built-in socks5 support in ipfs-desktop nor go-ipfs at this time. We are determined to improve the upgrade mechanism, so it does not break for users behind the Great Firewall, but for now you need to fix it manually.
> 
> I think the way to fix things for you is to:
> 
> 1. Manually download  the latest version of migration tool ([fs-repo-migrations_v1.7.1_windows-amd64.zip](https://ipfs.io/ipfs/QmYRLRDKobvg1AXTGeK5Xk6ntWTsjGiHbyNKhWfz7koGpa/fs-repo-migrations/v1.7.1/fs-repo-migrations_v1.7.1_windows-amd64.zip))  using some proxy
> 2. Add unpacked  `fs-repo-migrations.exe`   to your `PATH` – on Windows it means copying `.exe`  to `C:\Windows\system32\` or [manually adding some other directory via this interface](https://web.archive.org/web/20201223191048/https://www.architectryan.com/2018/03/17/add-to-the-path-on-windows-10/)
> 3. Quit IPFS Desktop and start it again
> 
> Let us know if this worked.
hello ^-^  I get the similar question but i cannot solve it.
I can open the version url like("https://ipfs.io/ipns/dist.ipfs.io/fs-repo-10-to-11/versions")
and I have download fs-repo-migrations,but I run that in term get error:
```
Could not get available repo migrations: http.DefaultClient.Do error: Get "https://ipfs.io/ipns/dist.ipfs.io/fs-repo-10-to-11/versions": dial tcp 54.89.135.129:443: connect: connection refused
```
my os is mac,and how can i solve that?tks
